pkg/util/purge/purge.go

```diff
@@ -0,0 +1,197 @@
+package purge
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+// all the purge functions are located here
+
+import (
+	"context"
+	"os"
+	"sort"
+
+	mgmtfeatures "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-07-01/features"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/graphrbac"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/authorization"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/features"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/network"
+	"github.com/Azure/ARO-RP/pkg/util/subnet"
+)
+
+type checkFn func(mgmtfeatures.ResourceGroup, *logrus.Entry) bool
+
+// ResourceCleaner hold the context required for cleaning
+type ResourceCleaner struct {
+	log    *logrus.Entry
+	dryRun bool
+
+	resourceRegions []string
+
+	resourcegroupscli      features.ResourceGroupsClient
+	vnetscli               network.VirtualNetworksClient
+	privatelinkservicescli network.PrivateLinkServicesClient
+	securitygroupscli      network.SecurityGroupsClient
+	roleassignmentcli      authorization.RoleAssignmentsClient
+	applicationscli        graphrbac.ApplicationsClient
+	serviceprincipalcli    graphrbac.ServicePrincipalsClient
+
+	subnetManager subnet.Manager
+
+	shouldDelete checkFn
+}
+
+// NewResourceCleaner instantiates the new RC object
+func NewResourceCleaner(log *logrus.Entry, shouldDelete checkFn, dryRun bool) (*ResourceCleaner, error) {
+	subscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	tenantID := os.Getenv("AZURE_TENANT_ID")
+
+	authorizer, err := auth.NewAuthorizerFromEnvironment()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	graphAuthorizer, err := auth.NewAuthorizerFromEnvironmentWithResource(azure.PublicCloud.GraphEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ResourceCleaner{
+		log:    log,
+		dryRun: dryRun,
+
+		resourceRegions: []string{"v4-eastus", "v4-westeurope", "v4-australiasoutheast"},
```

Comment:
Not used anymore. Please remove. We will need follow-up PR. Same wilt un-used clients bellow and in the `util`